README.md

```diff
@@ -1,2 +1,1088 @@
+  -p PRESET, --preset PRESET
+                        A maze+start+goals preset to load.
+  -e {default,better}, --engine {default,better}
+                        The render engine to use. (default: a simple pygame GUI renderer, requires the 'gui' feature)
+```
+
+(note that the new engine appeared as an option)
+
+## Tools
+
+This package also allows bundling various Micromouse-related tools.
+
+Tools are accessible via the `micromouse tool` command as subcommands:
+
+```sh
+$ micromouse tool -h
+usage: micromouse tool [-h] {maze} ...
+
+Run a tool.
+
+options:
+  -h, --help  show this help message and exit
+
+tools:
+  All registered tools.
+
+  {maze}
+    maze      Utilities for viewing and manipulating a maze.
+```
+
+### Provided Tool(s)
+
+#### Maze Editor
+
+This package provides a *MazeEditor* tool for basic creation and manipulation
+of mazes.
+
+```sh
+$ micromouse tool maze -h
+usage: micromouse tool maze [-h] {generate,render,rotate,transpose,flip} ...
+
+Utilities for viewing and manipulating a maze.
+
+options:
+  -h, --help            show this help message and exit
+
+subcommands:
+  {generate,render,rotate,transpose,flip}
+    generate            Generate a new maze.
+    render              Render a maze (can be used to convert other formats into .maze format).
+    rotate              Rotate a maze (can be used to convert other formats into .maze format).
+    transpose           Transpose a maze.
+    flip                Flip a maze.
+
+Loaded from sim.tools:MazeEditor
+```
+
+The avaliable commands for the *MazeEditor* tool are:
+
+##### `generate` - Generate a maze
+
+```sh
+$ micromouse tool maze generate -h
+usage: micromouse tool maze generate [-h] [-s SIZE] [--empty | --full] output_file
+
+Generate a new maze.
+
+positional arguments:
+  output_file           The path to save the new maze at ('-' for stdout).
+
+options:
+  -h, --help            show this help message and exit
+  -s SIZE, --size SIZE  The size of the maze to create. (default: 16x16)
+  --empty               Generate an empty maze (no walls).
+  --full                Generate a full maze (all walls). (default)
+```
+
+This command is useful for creating new mazes by creating a full maze and
+manually removing walls:
+
+```sh
+micromouse tool maze generate 16x16 > new.maze
+```
+
+##### `render` - Render a maze
+
+```sh
+$ micromouse tool maze render -h
+usage: micromouse tool maze render [-h] [-A] [-U] [--cell-width CELL_WIDTH] [--cell-height CELL_HEIGHT] [--no-force-corners] [-o OUTPUT_FILE] maze
+
+Render a maze.
+
+positional arguments:
+  maze                  The maze to load. May be a file or a maze input.
+
+options:
+  -h, --help            show this help message and exit
+  -A, --ascii           Use ASCII characters to draw the maze. (default)
+  -U, --unicode         Use Unicode (UTF-8) characters to draw the maze.
+  --cell-width CELL_WIDTH
+                        The amount of characters between to cell corners horizontally. (default: 3).
```

Comment:
```suggestion
                        The amount of characters between cell corners horizontally. (default: 3).
```

Reply:
```suggestion
                        The amount of characters between two cell corners horizontally. (default: 3).
```

Reply:
Pretty sure it was meant to be "two", but nothing works as well